drivers/samsung/displays/md_series.cr

```diff
@@ -0,0 +1,500 @@
+      bytes = io.peek
+      logger.debug { "Received: #{bytes}" }
+
+      # (data length + header and checksum)
+      # original ruby code with support for indicator and variable length message
+      # string[2].to_i + 4
+      bytes[3].to_i + 5 # move data length index + 1 as response will include indicator
+    end
+  end
+
+  def on_load
+    transport.tokenizer = init_tokenizer
+    on_update
+
+    self[:volume_min] = 0
+    self[:volume_max] = 100
+
+    # Meta data for inquiring interfaces
+    self[:type] = :lcd
+    self[:input_stable] = true
+    self[:input_target] ||= :hdmi
+    self[:power_stable] = true
+  end
+
+  def on_update
+    @id = setting(Int32, :display_id)
+    @rs232 = setting(Bool, :rs232_control)
+    @blank = setting(String?, :blank)
+  end
+
+  def connected
+    do_poll
+    do_device_config unless self[:hard_off]?
+
+    schedule.every(30.seconds) do
+      logger.debug { "-- polling display" }
+      do_poll
+    end
+  end
+
+  def disconnected
+    self[:power] = false unless @rs232
+    schedule.clear
+  end
+
+  # As true power off disconnects the server we only want to
+  # power off the panel. This doesn't work in video walls
+  # so if a nominal blank input is
+  # TODO: check type for broadcast is correct
+  def power(power : Bool, broadcast : String? = nil)
+    self[:power_target] = power
+    self[:power_stable] = false
+
+    if !power
+      # Blank the screen before turning off panel if required
+      # required by some video walls where screens are chained
+      switch_to(@blank.as(String)) if @blank && self[:power]?
+      do_send("panel_mute", 1)
+    elsif !@rs232 && !self[:connected]?
+       wake(broadcast)
+    else
+      # Power on
+      do_send("hard_off", 1)
+      do_send("panel_mute", 0)
+    end
+  end
+
+  def hard_off
+    do_send("panel_mute", 0) if self[:power]?
+    do_send("hard_off", 0)
+    do_poll
+  end
+
+  # TODO: figure out what block is for
+  # def power?(**options, &block)
+  def power?(**options)
+    # options[:emit] = block unless block.nil?
+    do_send("panel_mute", [] of UInt8, **options)
+  end
+
+  # Adds mute states compatible with projectors
+  def mute(state : Bool = true)
+    power(!state)
+  end
+
+  def unmute
+    power(true)
+  end
+
+  # check software version
+  def software_version?
+    do_send("software_version")
+  end
+
+  def serial_number?
+    do_send("serial_number")
+  end
+
+  # Converts a hex encoded string into a binary string
+  def byte_to_hex(data : String) : String
+    # Removes invalid characters
+    data = data.gsub(/(0x|[^0-9A-Fa-f])*/, "")
+
+    # Ensure we have an even number of characters
+    data = '0' + data if data.size % 2 > 0
+
+    # Breaks string into an array of characters
+    output = [] of Int32
+    data.scan(/.{2}/) do |md|
+      output << md[0].to_i(16)
+    end
+
+    String.build do |io|
+      output.each do |number|
+        io.write_byte number.to_u8
+      end
+    end
+  end
```

Comment:
This can be swapped for the [implementation available from Transcoder](https://github.com/PlaceOS/driver/blob/master/src/placeos-driver/utilities/transcoder.cr).

Reply:
Hey @KimBurgess , I had to make this change on my local to get this working but I've also opened a PR for it.

https://github.com/PlaceOS/driver/pull/32/files

Reply:
I've removed this as it's not actually needed